kratos/input_output/stl_io.cpp

```diff
@@ -217,7 +218,17 @@ void StlIO::ReadSolid(
     KRATOS_ERROR_IF(word != "solid") << "Invalid stl file. Solid block should begin with \"solid\" keyword but \"" << word << "\" was found" << std::endl;
     std::getline(*mpInputStream, word); // Reading solid name to be the model part name
 
-    word.erase(word.begin(), std::find_if(word.begin(), word.end(), [](int ch) {return !std::isspace(ch);})); // Triming the leading spaces
+    // Remove Comments
+    std::size_t pos = word.find("COMMENT:");
+    if(pos!=std::string::npos){
+        word = word.substr(0,pos);
+    }
+    pos = word.find(";");
+    if(pos!=std::string::npos){
+        word = word.substr(0,pos);
+    }
+    // Remove newline and spaces characters from string in C++
+    word = std::regex_replace( word, std::regex("\\r\\n|\\r|\\n| "), "");
```

Comment:
Copying my suggestion from the previous PR ...
(no regex, no reallocs)
```suggestion
for (const auto& symbol : {"COMMENT:", ";"}) {
        const auto position = word.find(symbol);
        if (position != word.npos) {
            word.erase(position);
        }
    }
    word.erase(std::remove_if(
        word.begin(),
        word.end(),
        [](auto character) -> bool {
            return character == '\r' || character == '\n';
        }
    ), word.end());
```